Userland/Applications/Assistant/main.cpp

```diff
@@ -277,8 +277,11 @@ int main(int argc, char** argv)
         GUI::Application::the()->quit();
     };
     window->on_active_window_change = [](bool is_active_window) {
-        if (!is_active_window)
-            GUI::Application::the()->quit();
+        if (!is_active_window) {
+            auto app = GUI::Application::the();
+            app->quit();
+            app->event_loop().pump();
```

Comment:
This looks extremely suspicious. Why isn't the event loop being pumped normally by `Core::EventLoop::exec()`?

Reply:
I am wondering too. However I just noticed that I tested another solution, which did work but I did not rebuild with those changes.
```
app->set_quit_when_last_window_deleted(true);
window->on_active_window_change = [&window](bool is_active_window) {
        if (!is_active_window)
            window->close();
    };
 ```
This works but I feel thats even worse.